client/client-multi/client-cpp/impl/utils/ThreadPool.cpp

```diff
@@ -114,32 +119,26 @@ void ThreadPool::shutdownNow()
 void ThreadPool::start()
 {
     for (std::size_t i = 0; i < workerCount_; ++i) {
-        workers_.push_back(std::thread(Worker(*this)));
+        workers_.emplace_back(Worker(*this));
     }
+
+    state_ = State::RUNNING;
 }
 
 void ThreadPool::stop(bool force)
```

Comment:
Consider making two distinct functions: `stop` and `force_stop` as they are different cases and seeing `stop(false)` is non-intuitive.